src/Authentication/Authentication/Cmdlets/ConnectMgGraph.cs

```diff
@@ -201,15 +220,26 @@ private async Task ProcessRecordAsync()
                     // Save auth context to session state.
                     GraphSession.Instance.AuthContext = await Authenticator.AuthenticateAsync(authContext, ForceRefresh, _cancellationTokenSource.Token);
                 }
-                catch(Exception ex)
+                catch (Exception ex)
                 {
+                    if (IsUnableToOpenWebPageError(ex))
+                    {
+                        WriteWarning(Resources.InteractiveAuthNotSupported);
+                        WriteDebug(ex.ToString());
+                    }
```

Comment:
Should we just warn customers that interactive auth is not supported and switch to device code without throwing an exception?

The current design would result in a breaking change for customers who currently use our SDK in environments where browsers are not available i.e. CloudShell, Linux servers, Azure functions e.t.c.

Reply:
Taking inspiration from [Azure](https://github.com/Azure/azure-powershell/blob/22119cadba08bba03c5f596f92aa7e9bbd7fa05e/src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs#L441-L464), where they inform the customer that Interactive is not supported and requiring them to make the conscious choice to switch to device code.

Reply:
We may have to take a different tangent with this to avoid breaking our customers' scripts. The current approach will force customers running the SDK in non-browser environments to update their scripts and add `-UseDeviceAuthentication`. This is fine if we are also bumping the major version to signify a breaking change; otherwise, we should fall back to the current behavior if we can't open a browser.

@ddyett, @darrelmiller, thoughts on this?

Reply:
![image](https://user-images.githubusercontent.com/1641829/113527048-ebf89e80-95c4-11eb-804b-7f5887b71f12.png)
Shown above, Warning to the user about the fallback.